pkg/api/datasources.go

```diff
@@ -69,16 +71,39 @@ func GetDataSourceById(c *middleware.Context) Response {
 }
 
 func DeleteDataSource(c *middleware.Context) {
```

Comment:
Since we should have two urls for deleting datasources we should also splitt this handler into two.